ext/storage_sources/s3_store/s3_storage_source.cpp

```diff
@@ -685,6 +744,10 @@ wiredtiger_extension_init(WT_CONNECTION *connection, WT_CONFIG_ARG *config)
         return (ret != 0 ? ret : EINVAL);
     }
 
+    /* Set up statistics. */
+    if ((s3->statistics = (S3_STATISTICS *)calloc(1, sizeof(S3_STATISTICS))) == NULL)
```

Comment:
Is there a good reason to dynamically allocate statistics?
How about the statistics structure be made part of the storage source, not the pointer to the stats.